packages/cli/src/commands/generate/schema-from-database/generate_schema_command.ts

```diff
@@ -54,9 +53,8 @@ export class GenerateSchemaCommand
   handler = async (
     args: ArgumentsCamelCase<GenerateSchemaCommandOptions>
   ): Promise<void> => {
-    const backendIdentifier = await this.backendIdentifierResolver.resolve(
-      args
-    );
+    const backendIdentifier =
+      await this.backendIdentifierResolver.resolveBackendIdentifier(args);
 
     if (!backendIdentifier) {
       throw new AmplifyFault('BackendIdentifierFault', {
```

Comment:
Should this be a fault if the input is coming from the user?

Reply:
I think this should stay as a fault because if we can't resolve using input from the user we fallback to resolving to sandbox backend ID. The only thing really controlled by the user at that point is if they have the `name` field in their `package.json` which is covered here https://github.com/aws-amplify/amplify-backend/blob/main/packages/cli/src/backend-identifier/local_namespace_resolver.ts#L27.

Reply:
This is concerning. if a user provided a `--stack` argument that doesn't result in a valid identifier, why would we fallback to sandbox backend identifier? This seems like a wrong fallback when customer intention is clear.

Reply:
True... this fallback is there for all generate commands unfortunately

Reply:
I can follow up with a PR to not fallback if user provided stack or app ID and branch then change this to an error

Reply:
Sounds good. I think in some context it might makes sense to have this fallback but maybe not in the cli commands.